src/server/pkg/hashtree/interface.go

```diff
@@ -100,6 +100,9 @@ type OpenHashTree interface {
 	// PutFile appends data to a file (and creates the file if it doesn't exist).
 	PutFile(path string, objects []*pfs.Object, size int64) error
 
+	// PutFileSplit appends data to a file (and creates the file if it doesn't exist).
+	PutFileSplit(path string, objects []*pfs.Object, size int64, header *pfs.Object, footer *pfs.Object, headerFooterSize int64) error
```

Comment:
A bit on the fence about this signature / name. Adding 3 more args seems heavy - we could just create a struct for `metadata` that includes the header/footer/size ... and then call this method `PutFileWithMetadata`